### PR TITLE
Created New note UI and added Yup validations for the Notes form

### LINKED ISSUE #4: Create new note UI

### PR DESCRIPTION
**Pull request by LiyakathMohammed**
Fixes #4

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Notes/Card.jsx [hunk 1]
```diff
@@ -20,15 +20,20 @@ const Card = ({ title, description, assignedContactId, tag, created_at }) => {
             {description}
           </p>
           <div className="flex w-full items-center justify-between">
-            <Tag className="mt-2" label={tag} />
+            <Tag className="mt-2" label={tag.value} />
             <p className="mt-2 flex items-center text-sm text-gray-600">
               <Clock className="mr-1 h-4 w-4" />
-              {moment.utc(created_at).local().startOf("seconds").fromNow()}
+              <Tooltip
+                content={moment(created_at).format("dddd, h:mm a")}
```
Comment (navaneethsdk) [inline app/javascript/src/components/Dashboard/Notes/Card.jsx:27]: Move the formatting to a separate utils function with a proper name.
Also, use `dayjs` over other libraries and inbuilt functions: https://courses.bigbinaryacademy.com/learn-react/react-best-practices/miscellaneous-tips/#prefer-dayjs-over-inbuilt-date-functions

File: app/javascript/src/components/Dashboard/Notes/Card.jsx [hunk 2]
```diff
@@ -20,15 +20,20 @@ const Card = ({ title, description, assignedContactId, tag, created_at }) => {
...
+                position="bottom"
+              >
+                {moment.utc(created_at).local().startOf("seconds").fromNow()}
```
Comment (navaneethsdk) [inline app/javascript/src/components/Dashboard/Notes/Card.jsx:30]: Same as above comment.

File: app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx [hunk 3]
```diff
@@ -1,26 +1,29 @@
...
       {
-        title,
-        description,
-        assignedContactId: assignedContact.value,
-        tag: tag.value,
+        title: values.title,
+        description: values.description,
+        assignedContact: { ...values.assignedContact },
+        tag: { ...values.tag },
+        created_at: moment().format(),
```
Comment (navaneethsdk) [inline app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx:21]: Destructure the values here.

File: app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx [hunk 4]
```diff
@@ -63,6 +66,8 @@ const Form = ({ onClose, note, isEdit, setShowPane, notes, setNotes }) => {
             />
             <div className="w-full space-y-0">
               <Select
+                isClearable
```
Comment (navaneethsdk) [inline app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx:69]: Isn't `isClearable` true by default?

**Review by navaneethsdk — COMMENTED** [review]
@LiyakathMohammed a few comments

**Comment by LiyakathMohammed** [comment]
@navaneethsdk 

I have moved the formatting and elapsed time to utils and destructured the form values as I recieve.

If I remove isClearable, then I am unable to clear my option. So I kept it as it is.
Should I remove?

File: app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx [hunk 5]
```diff
@@ -1,26 +1,29 @@
...
       ...notes,
       {
-        title,
-        description,
-        assignedContactId: assignedContact.value,
-        tag: tag.value,
+        title: values.title,
+        description: values.description,
+        assignedContact: { ...values.assignedContact },
+        tag: { ...values.tag },
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx:20]: Can you explain to me why we are spreading and creating a new object here?

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 6]
```diff
@@ -3,25 +3,58 @@ import * as yup from "yup";
 export const NOTES_FORM_INITIAL_VALUES = {
   title: "",
   description: "",
-  assignedContactId: "",
-  tag: "",
+  assignedContact: [],
+  tag: [],
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js:7]: How does assigning a singular entity, that is `tag`, with a collection like array make sense? If it's something that can hold a collection then name it appropirately.

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 7]
```diff
@@ -3,25 +3,58 @@ import * as yup from "yup";
 export const NOTES_FORM_INITIAL_VALUES = {
   title: "",
   description: "",
-  assignedContactId: "",
-  tag: "",
+  assignedContact: [],
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js:6]: Same as last comment.

**Review by yedhink — COMMENTED** [review]
@LiyakathMohammed

File: app/javascript/src/components/Dashboard/Notes/constants.js (hunk 7 above)
Comment (LiyakathMohammed) [reply app/javascript/src/components/Dashboard/Notes/constants.js:6]: @yedhink, If I change it to empty string, I am getting this following error

`Error: Objects are not valid as a React child (found: object with keys {label, value}). If you meant to render a collection of children, use an array instead.`

It was originally empty, but as I was getting this error, I tried different solutions from web but only changing the Initital values to array helped me solve this error.

File: app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx (hunk 5 above)
Comment (LiyakathMohammed) [reply app/javascript/src/components/Dashboard/Notes/Pane/Form/index.jsx:20]: This was originally

`{
        title,
        description,
        assignedContact,
        tag,
        createdAt: dayjs(),
      }`
      
 For the mentioned error in the below comment, I was changing a few values and adding a few to see where exactly is the error. I kept it like it until adding the delete feature.
 
 I changed it in the commit related to delete feature.

File: app/javascript/src/components/Dashboard/Notes/constants.js (hunk 7 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/constants.js:6]: @LiyakathMohammed https://www.loom.com/share/99d16c33e41041ee87db966ece9d27a5

Comment (LiyakathMohammed) [reply app/javascript/src/components/Dashboard/Notes/constants.js:6]: Fixed it by assigning both assignedTo and tag to null.

**Pull request merged**